### ISSUE TITLE
AsymmetricNetworkPartitionIT.shouldWithstandAsymmetricNetworkPartition

### ISSUE THREAD

**Issue by npepinpe** [camunda/zeebe#9266]
**Summary**

- How often does the test fail? at least twice
- Does it block your work? no
- Do we suspect that it is a real failure? I don't think so

**Failures**

<details><summary>Example assertion failure</summary>
<pre>
java.lang.AssertionError: 

Expecting actual:
  0
not to be equal to:
  0

	at io.camunda.zeebe.it.clustering.network.AsymmetricNetworkPartitionIT.shouldWithstandAsymmetricNetworkPartition(AsymmetricNetworkPartitionIT.java:77)
</pre>
</details>

**Hypotheses**

None so far

**Logs**

None available. Failing build is: https://github.com/camunda/zeebe/actions/runs/2256830107

**Comment by Zelldon** [comment]
Not easy reproducible.

**Comment by Zelldon** [comment]
Ok taking a look at the code it is clear what the issue is. https://github.com/camunda/zeebe/blob/main/qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/network/AsymmetricNetworkPartitionIT.java#L77

We can't guarantee that the leaders are spreaded over all nodes, but since the test doesn't work if the partition leader is on the same node we fail here. We maybe need to provoke a leader change instead.

**Comment by npepinpe** [comment]
Could we use priority election here?

**Comment by npepinpe** [comment]
It's enabled, but if you don't assign priorities then I guess it behaves as if it wasn't enabled :smile:

**Comment by Zelldon** [comment]
Ah good to know :laughing: I though we assign already some good defaults

**Comment by npepinpe** [comment]
I'm trying to figure out now actually what would be the default priority. Maybe we do something with it for all I know, it's been a while since I looked into that.

**Comment by Zelldon** [comment]
>  I guess that explains why it almost always works but not quite

Yeah exactly :D

**Comment by npepinpe** [comment]
We could use the rebalance API until we get the distribution we want. I think it's also not guaranteed, but it works with the priorities, such that the other nodes will delay starting an election if they don't have the highest priority.

**Comment by npepinpe** [comment]
Still flaky, but with a different error now. I think the condition to rebalance is not robust enough (it seems there's some null pointer, ups :smile:)

**Comment by npepinpe** [comment]
Yeah, makes a lot of sense. We rebalance then immediately try to find who is the new leader, which of course is not so quick at times, so we have to be a bit more lenient on that.
